### PR TITLE
Add docker containers and release actions 

### PR DESCRIPTION
**Pull request by uncleDecart**

### CODE REVIEW

**Comment by uncleDecart** [comment]
Solves #16, we will have to merge and create release to debug GHA

**Review by deitch — COMMENTED** [review]
Is there any reason not to have a single container image with both binaries in it? They are pretty small, it gives you everything in a single image, makes the repo cleaner with a single Dockerfile.

I am pretty sure etcd, redis, mysql, mariadb, postgres, etc. all do it the same way.

**Comment by uncleDecart** [comment]
Then your entry point will be shell and then you write nkv-server "0.0.0.0:4222" or nkv-client "127.0.0.1:2222", okay that makes sense, I'll group it all in :) Thanks, Avi!

**Comment by uncleDecart** [comment]
@deitch this too

**Comment by uncleDecart** [comment]
Rebased to master so that Cargo.lock is a valid file

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
+ARG RUST_VERSION=1.80.1
+
+FROM rust:${RUST_VERSION}-alpine3.20 AS toolchain
+ENV CARGO_BUILD_TARGET="x86_64-unknown-linux-musl"
```
Comment (deitch) [inline Dockerfile:4]: Why only x86_64? Wouldn't you want to build for arm64 as well? Maybe riscv64 😁

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+RUN rustup target add ${CARGO_BUILD_TARGET}
+RUN apk add --no-cache musl-dev linux-headers make clang mold
+ADD config.toml /usr/local/cargo/
+# CARGO_BUILD_TARGET is respected by cargo install and other cargo commands
+RUN cargo install cargo-chef@0.1.67
+
+FROM toolchain AS planer
```
Comment (deitch) [inline Dockerfile:11]: I am not sure I get this. There are 3 different stages: cache, "planer" (what is this?) and builder. I get builder, but what are the purposes of cache and "planer"?

File: README.md [hunk 3]
```diff
@@ -20,7 +20,27 @@ For more information reffer to [this](./docs/DESIGN_DECISIONS.md)
...
+You can directly pull docker containers for client and server. They are published with each release:
+
+Make sure that you have docker [installed](https://docs.docker.com/engine/install/)
+
+```sh
+docker run -d --net=host uncledecart/nkv:latest ./nkv-server "0.0.0.0:4222"
+docker run -it --net=host uncledecart/nkv:latest ./nkv-client "4222"
+```
+
+Note that when using network different from host you might face major performance degradation, also docker
```
Comment (deitch) [inline README.md:35]: That is a bit harsh. Maybe tone it down? 

> when using network other than the host, the network implementation may impact network performance, unrelated to nkv itself.

**Review by deitch — COMMENTED** [review]
Some questions and a suggestion.

File: Dockerfile (hunk 1 above)
Comment (uncleDecart) [reply Dockerfile:4]: "Do things that don't scale" Make it work for x86_64 and then expand to other platforms.

Comment (deitch) [reply Dockerfile:4]: Fine.

**Review by deitch — APPROVED** [review]

**Pull request merged**